### ISSUE TITLE
Masked TextInput does not update displayed value in relation edit modal

### ISSUE THREAD

**Issue by henninghorn** [laravel-filament/filament#1289]
### Package

filament/filament

### Package Version

v2.9.7

### Laravel Version

v8.80.0

### PHP Version

PHP 8.1

### Bug description

I have the following form on a relation manager:

```php
public static function form(Form $form): Form
{
    return $form
        ->schema([
            TextInput::make('name'),
            Textarea::make('description'),
            TextInput::make('price')
                ->postfix('kr.')
                ->mask(fn (TextInput\Mask $mask) => $mask
                ->patternBlocks([
                    'money' => fn (Mask $mask) => $mask
                        ->numeric()
                        ->integer()
                        ->thousandsSeparator('.')
                ])
                ->pattern('money')),
        ]);
}
```

When I click edit on the first item, I get this modal:

![billede](https://user-images.githubusercontent.com/742607/150085510-7f1ea3ca-a310-4242-880a-828ad4a920f9.png)

Whether I update the price field or not, that value it has when the modal is closed, is "transferred"/"persisted" to the next edit modal.

![billede](https://user-images.githubusercontent.com/742607/150084714-3fc0d5c8-c376-4104-96a3-d08ba58789bf.png)

The actual price should be 3000 (which is the value stored in the database), and not 1234, as the other package was updated to. However, if I click save, the price is not updated to the price that is displayed. So the problem seems to be front end related.

### Steps to reproduce

Step 1 - Create a relation manager for a resource

Step 2 - Create a masked TextInput field for the related resource
```php
TextInput::make('price')
    ->mask(fn (TextInput\Mask $mask) => $mask
    ->patternBlocks([
        'money' => fn (Mask $mask) => $mask
            ->numeric()
            ->integer()
            ->thousandsSeparator('.')
    ])
    ->pattern('money')),
```

Step 3 - Create two or more related resource, so a list of 2+ resources is displayed

Step 4 - Click edit on the first resource, then close modal and click edit on the second

The masked value from the first modal is "persisted" between any of the other edit modals until full page refresh.

**Comment by danharrin** [comment]
Any ideas how we can reload the mask reliably when the value changes @ryangjchandler? Or is this just a limitation with iMask?

**Comment by danharrin** [comment]
Unfortunately I think this is a limitation with iMask - it does not support changing the value after the input is loaded.

However - I'm looking into creating a MoneyInput which uses a different library and is more stable 👍

**Comment by danharrin** [comment]
Fixed by #1533.

**Comment by danharrin** [comment]
Ah, thats a real shame. I assumed it did, since it does fix setting the value to `null` -  which is where I thought this was going wrong (when switching records) 😔
